roofit/roofitcore/inc/RooNormSetCache.h

```diff
@@ -16,28 +16,32 @@
 #ifndef ROO_NORMSET_CACHE
 #define ROO_NORMSET_CACHE
 
+#include <RooArgSet.h>
+
+#include <Rtypes.h>
+
 #include <vector>
 #include <map>
 #include <string>
 
-#include "Rtypes.h"
-
-class RooAbsArg;
-class RooArgSet;
-
 typedef RooArgSet* pRooArgSet ;
 
 class RooNormSetCache {
 protected:
-  typedef std::pair<const RooArgSet*, const RooArgSet*> Pair;
-  struct PairCmp {
-      inline bool operator()(const Pair& a, const Pair& b) const
-      {
-     if (a.first < b.first) return true;
-     if (b.first < a.first) return false;
-     return a.second < b.second;
-      }
+  class Pair {
+  public:
+    using Value_t = RooFit::UniqueId<RooArgSet>::Value_t;
+    Pair(const RooArgSet* set1, const RooArgSet* set2)
+      : _pair{RooFit::getUniqueId(set1), RooFit::getUniqueId(set2)} {}
+   bool operator==(Pair const &other) const { return _pair == other._pair; }
+   bool operator<(Pair const &other) const { return _pair < other._pair; }
+
+   Value_t const& first() const { return _pair.first; }
+   Value_t const& second() const { return _pair.second; }
+  private:
+    std::pair<Value_t,Value_t> _pair;
   };
+
   typedef std::vector<Pair> PairVectType;
   typedef std::map<Pair, ULong_t> PairIdxMapType;
```

Comment:
Should we use here ULong64_t ?

Reply:
Yes ok, it's true that the fixed size integers are more concise.